packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -290,6 +312,31 @@ const BUTTON_INPUT = [InputStruct, ButtonStruct] as [
   typeof ButtonStruct,
 ];
 
+/**
+ * A subset of JSX elements that represent the tuple Box + Input of the Field children.
+ */
+const BOX_INPUT_LEFT = [BoxStruct, InputStruct] as [
+  typeof BoxStruct,
+  typeof InputStruct,
+];
+
+/**
+ * A subset of JSX elements that represent the tuple Input + Box of the Field children.
+ */
+const BOX_INPUT_RIGHT = [InputStruct, BoxStruct] as [
+  typeof InputStruct,
+  typeof BoxStruct,
+];
+
+/**
+ * A subset of JSX elements that represent the tuple Box + Input + Box of the Field children.
+ */
+const BOX_INPUT_BOTH = [BoxStruct, InputStruct, BoxStruct] as [
```

Comment:
I think we should probably just use `BoxChildStruct` instead. That includes `Box`, but will also let you use `Button`. Then we can remove `BUTTON_INPUT`.

Reply:
I do not see a way how to make it work with `FieldStruct` because of the `FieldProps -> children` which expects exact elements or tuple. I tried couple of things, but it doesn't seem to work. Let me know if you have working solution.

Reply:
We can use `GenericSnapElement` instead of `BoxElement` for the field `children` type